apps/browser/src/popup/app.component.ts

```diff
@@ -60,6 +60,12 @@ export class AppComponent implements OnInit, OnDestroy {
       this.activeUserId = userId;
     });
 
+    this.stateService.activeAccountUnlocked$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((unlocked) => {
+        unlocked ? this.recordActivity() : null;
+      });
```

Comment:
Unfortunately `recordActivity` is async, which makes the subscription way more annoying. In general, we use [`concatMap`](https://www.learnrxjs.io/learn-rxjs/operators/transformation/concatmap) to handle awaiting things we need to subscribe to.

I think that's what we want here, too, because the order of the execution matters -- we don't want an earlier event to overwrite a later timestamp.

```suggestion
    this.stateService.activeAccountUnlocked$
      .pipe(
        filter((unlocked) => unlocked),
        concatMap(async () => {
          await this.recordActivity();
        }),
        takeUntil(this.destroy$)
      )
      .subscribe();
```